pkg/cloud/packet/client.go

```diff
@@ -144,23 +147,28 @@ func (p *Client) NewDevice(ctx context.Context, req CreateDeviceRequest) (*packn
 		facility = req.MachineScope.PacketMachine.Spec.Facility
 	}
 
-	serverCreateOpts := &packngo.DeviceCreateRequest{
-		Hostname:      req.MachineScope.Name(),
-		ProjectID:     req.MachineScope.PacketCluster.Spec.ProjectID,
-		Facility:      []string{facility},
-		BillingCycle:  req.MachineScope.PacketMachine.Spec.BillingCycle,
-		Plan:          req.MachineScope.PacketMachine.Spec.MachineType,
-		OS:            req.MachineScope.PacketMachine.Spec.OS,
-		IPXEScriptURL: req.MachineScope.PacketMachine.Spec.IPXEUrl,
-		Tags:          tags,
-		UserData:      userData,
+	hostname := req.MachineScope.Name()
+	serverCreateOpts := metal.CreateDeviceRequest{
+		DeviceCreateInFacilityInput: &metal.DeviceCreateInFacilityInput{
+			Hostname: &hostname,
+			Facility: metal.FacilityInputFacility{
```

Comment:
Setting aside the weirdness of the names here:  the `FacilityInputFacility` model seems to serialize incorrectly, so we may need to copy the `metal-java` [patch that only supports string arrays for facility](https://github.com/equinix-labs/metal-java/blob/main/patches/spec.fetched.json/05-remove-because-java-generator-lack-of-support-for-primitive-anyof.patch).  It looks like facility is being posted as `"facility":{"ArrayOfString":null,"String":"fr2"}`

Reply:
This was fixed by equinix-labs/metal-go#90